pilot/lib/pilot/build_manager.rb

```diff
@@ -389,15 +389,19 @@ def expire_previous_builds(build)
     def transporter_for_selected_team(options)
       # Use JWT auth
       api_token = Spaceship::ConnectAPI.token
+      api_key = options[:api_key]
+      # Load api key info if user set api_key_path, not api_key
+      api_key = { key_id: api_token.key_id, issuer_id: api_token.issuer_id, key: api_token.key_raw, is_key_content_base64: api_token.is_key_content_base64 } if api_key.nil? && !options[:api_key_path].nil?
```

Comment:
You can use `||=` instead to set default value.

```suggestion
      api_key ||= { key_id: api_token.key_id, issuer_id: api_token.issuer_id, key: api_token.key_raw, is_key_content_base64: api_token.is_key_content_base64 }
```

Reply:
[In my opinion] It's better to use `Struct` instead of Hash to describe `api_key`.
However, I'm concerned about the compatibility between existing APIs.

Reply:
@giginet I think using `||=` would not have the same behavior here because in the original code there's also a condition on `&& !options[:api_key_path].nil?`

Another alternative to write those lines could also be:
```
api_key = if options[:api_key].nil? && !options[:api_key_path]
  …hash value here…
elsif …
  …
else
  …
end
```
(Though weather that would be more readable than the original code is debatable and a matter of preference)